dis_snek/api/events/processors/channel_events.py

```diff
@@ -40,7 +39,7 @@ async def _on_raw_channel_update(self, event: "RawGatewayEvent") -> None:
     @Processor.define()
     async def _on_raw_channel_pins_update(self, event: "RawGatewayEvent") -> None:
         channel = await self.cache.get_channel(event.data.get("channel_id"))
-        channel.last_pin_timestamp = timestamp_converter(event.data.get("last_pin_timestamp"))
+        channel.last_pin_timestamp = event.data.get("last_pin_timestamp")
```

Comment:
My gut says you're changing this from a `datetime` to a `str`

Reply:
```suggestion
        if last_pin_timestamp := event.data.get("last_pin_timestamp"):
            channel.last_pin_timestamp = timestamp_converter(last_pin_timestamp)
```

I think something like this make more sense

Reply:
I think it's ok because attrs will convert this automatically 
![](https://i.imgur.com/Z68OTYo.png)

Reply:
The last time i read attrs docs, the converter only happens on init, not property set. Have you tested it?

Reply:
> The last time i read attrs docs, the converter only happens on init, not property set. Have you tested it?

Yea, I'll try again

Reply:
Converter works on set!

Reply:
that is odd . I suspect the json parsing converts datetime for you.

![image](https://user-images.githubusercontent.com/30431861/150983624-135e25a3-f7ad-4436-93ed-9a9e2b60c97c.png)

attrs docs itself quotes...
> Please note that converters only run on initialization.

Reply:
> that is odd . I suspect the json parsing converts datetime for you.
> 
> ![image](https://user-images.githubusercontent.com/30431861/150983624-135e25a3-f7ad-4436-93ed-9a9e2b60c97c.png)
> 
> attrs docs itself quotes...
> 
> > Please note that converters only run on initialization.

Test it yourself!

Reply:
![image](https://user-images.githubusercontent.com/30431861/150984298-a912addb-6829-4bc8-aed9-738a3e6d369a.png)
sure

Reply:
> ![image](https://user-images.githubusercontent.com/30431861/150984298-a912addb-6829-4bc8-aed9-738a3e6d369a.png) sure

I was testing this on real update pin timestamp in dis_snek, try it, please

Reply:
Guys, don't fight
Attr runs converters and validators only on init *by default*
You can also enable this behavior on `setattr` too when you define your attr model
[Here](https://github.com/Discord-Snake-Pit/Dis-Snek/blob/master/dis_snek/client/utils/attr_utils.py#L22) we enable this behavior for all of our models defined using our `@attr_utils.define` decorator

Reply:
ah i recall what we did now lol... mb


![image](https://user-images.githubusercontent.com/30431861/150985115-cc0cf7dd-9721-4e59-acb5-2fd739397342.png)

we added this

Reply:
remind me to not review code when I'm halfway through a school report haha